examples/cfd/ahmed_body_mgn/utils.py

```diff
@@ -15,34 +15,96 @@
 # limitations under the License.
 
 import torch
+import numpy as np
+import pyvista as pv
+import shapely
 
 
-def compute_drag_coefficient(normals, area, coeff, p, s):
+def compute_frontal_area(mesh: pv.PolyData, direction: str = "x"):
+    """
+    Compute frontal area of a given mesh
+    Ref: https://github.com/pyvista/pyvista/discussions/5211#discussioncomment-7794449
+
+    Parameters:
+    -----------
+    mesh: pv.PolyData
+        Input mesh
+    direction: str, optional
+        Direction to project the area. Defaults to "x".
+
+    Raises:
+    -------
+    ValueError: Only supports x, y and z projection for computing frontal area.
+
+    Returns:
+    --------
+    frontal area: float
+        Frontal area of the mesh in the given direction
+    """
+    direction_map = {
+        "x": ((1, 0, 0), [1, 2]),
+        "y": ((0, 1, 0), [0, 2]),
+        "z": ((0, 0, 1), [0, 1]),
+    }
+
+    if direction not in direction_map:
+        raise ValueError("Direction must be x, y or z only")
+
+    normal, indices = direction_map[direction]
+    areas_proj = mesh.project_points_to_plane(origin=(0, 0, 0), normal=normal)
+    merged = shapely.union_all(
+        [
+            shapely.Polygon(
+                np.stack(
+                    [
+                        areas_proj.points[tri, indices[0]],
+                        areas_proj.points[tri, indices[1]],
+                    ],
+                    axis=1,
+                )
+            )
+            for tri in areas_proj.triangulate().regular_faces
+        ]
+    )
+
+    return merged.area
+
+
+def compute_drag_coefficient(
```

Comment:
We discussed about adding a function for calculating lift. Another quantity of interest is the skin friction coefficient. It will be great if we can add that as it will also be useful for internal flow usecases. It is basically the local wall shear stress divided by the free stream kinetic energy and is calculated on each facet.

Reply:
Thanks @RishikeshRanade ! I modified the compute force calc utility to take arbitrary direction - now it can be used to compute drag or lift. Let me look into adding something for the skin friction

Reply:
Yes, its basically c_f but normalized by the free-stream kinetic energy. It would be great if we can split those 2, it will make them reusable for other applications.

Reply:
Thanks! modified the utility to return cp and cf separately. Resolving this thread now